invenio_records_resources/references/entity_resolvers/results.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Resolver for result items."""
+
+from invenio_access.permissions import system_identity
+
+from .base import EntityProxy, EntityResolver
+
+
+class ResultItemProxy(EntityProxy):
+    """Resolver proxy for a result item entity."""
+
+    def __init__(self, resolver, ref_dict, service):
+        """Constructor."""
+        super().__init__(resolver, ref_dict)
+        self.service = service
+
+    def _resolve(self):
+        """Resolve the result item from the proxy's reference dict."""
+        id_ = self._parse_ref_dict_id()
+        return self.service.read(system_identity, id_).to_dict()
```

Comment:
minor: not bocking, but could we make this identity configurable (i.e. passed from the `ResultItemResolver` to the `proxy_cls`)? It's a bit "dangerous" to have `system_identity` used like this since one might accidentally use it to get unauthorized information.

Reply:
Discussed IRL, and this is fine for now.